core/services/relay/evm/config_poller.go

```diff
@@ -105,18 +53,22 @@ type configPoller struct {
 	// contract allows us work around such restrictions.
 	configStoreContractAddr *common.Address
 	configStoreContract     *ocrconfigurationstoreevmsimple.OCRConfigurationStoreEVMSimple
+
+	// Depending on the exact contract used, the raw config log may be shaped
+	// in different ways
+	ld LogDecoder
 }
 
 func configPollerFilterName(addr common.Address) string {
 	return logpoller.FilterName("OCR2ConfigPoller", addr.String())
 }
 
-func NewConfigPoller(lggr logger.Logger, client client.Client, destChainPoller logpoller.LogPoller, aggregatorContractAddr common.Address, configStoreAddr *common.Address) (evmRelayTypes.ConfigPoller, error) {
```

Comment:
this signature is getting unwieldy. since you are making a break signature change anyway, i would make a config struct to house much of this stuff. maybe ~ NewConfigPoller(lggr, cfg)